project/frontend/src/components/Authentication.js

```diff
@@ -0,0 +1,54 @@
+import React, { useState, useEffect } from 'react';
+
+const AuthContext = React.createContext({});
+
+function Authentication(props) {
+  const [GoogleAuth, setGoogleAuth] = useState(undefined);
+  const [currentUser, setCurrentUser] = useState(undefined);
+  const [signedIn, setSignedIn] = useState(false);
+
+  const context = {
+    GoogleAuth: { get: GoogleAuth, set: setGoogleAuth },
+    currentUser: { get: currentUser, set: setCurrentUser },
+    signedIn: { get: signedIn, set: setSignedIn },
+  };
+
+  useEffect(() => {
+    const onGoogleAuthLoaded = () => {
+      window.gapi.load('auth2', () => {
+        const auth2 = window.gapi.auth2.init({
+          client_id:
+            '21423804760-e3goj1cdhg49ojdf780mcq92qgshbr4v.apps.googleusercontent.com',
```

Comment:
Let's put this in a gitignore file and import it as a variable

Reply:
Going to implement this .env file in a different PR.

Reply:
Sorry, commented this again below before I saw this!

Sometimes when api keys are added on github, the service like flags it.  So you probably want to submit the .env PR before you submit this one.